mobile/test/common/integration/client_integration_test.cc

```diff
@@ -71,6 +81,44 @@ void ClientIntegrationTest::basicTest() {
 
 TEST_P(ClientIntegrationTest, Basic) { basicTest(); }
 
+TEST_P(ClientIntegrationTest, ClearTextNotPermitted) {
+  EXPECT_CALL(helper_handle_->mock_helper(), isCleartextPermitted(_)).WillRepeatedly(Return(false));
+
+  expect_data_streams_ = false;
+  initialize();
+
+  Buffer::OwnedImpl request_data = Buffer::OwnedImpl("request body");
+  default_request_headers_.addCopy(AutonomousStream::EXPECT_REQUEST_SIZE_BYTES,
+                                   std::to_string(request_data.length()));
+
+  stream_prototype_->setOnData([this](envoy_data c_data, bool end_stream) {
+    if (end_stream) {
+      EXPECT_EQ(Data::Utility::copyToString(c_data), "Cleartext is not permitted");
+    }
+    cc_.on_data_calls++;
+    release_envoy_data(c_data);
+  });
+
+  stream_->sendHeaders(envoyToMobileHeaders(default_request_headers_), true);
+
+  terminal_callback_.waitReady();
+
+  ASSERT_EQ(cc_.on_headers_calls, 1);
+  ASSERT_EQ(cc_.status, "400");
+  ASSERT_EQ(cc_.on_data_calls, 1);
+  ASSERT_EQ(cc_.on_complete_calls, 1);
+
+  EXPECT_EQ(0, cc_.final_intel.stream_start_ms);
```

Comment:
it seems like the timing stats aren't exepected to be recorded, in which case, perhaps the checks here and below aren't needed?

Reply:
Yeah, good point. I debated this myself. Removed!